src/woo_search/search_index/typing.py

```diff
@@ -0,0 +1,15 @@
+from datetime import datetime
+from typing import Optional, TypedDict
+
+
+class DocumentType(TypedDict):
+    uuid: str
+    publicatie: str
+    publisher: str
+    identifier: Optional[str]
```

Comment:
I'd opt for using `default=""` on the serializer so that we can remove the `Optional`

```suggestion
    identifier: str
```

Allowing `None` values for string fields (serializer/model) is just weird and we can make life simpler